src/unwinder/i386/unwinder_i386.cpp

```diff
@@ -105,8 +105,19 @@ OOPStackUnwinderX86::VirtualUnwind(
     ENUM_CALLEE_SAVED_REGISTERS();
 #undef CALLEE_SAVED_REGISTER
 
-    ContextRecord->Esp = rd.SP - codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
-    ContextRecord->ResumeEsp = rd.SP;
+    SIZE_T paramSize = codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
+    SIZE_T paddingSize = 0;
+
+#ifdef UNIX_X86_ABI
+    // On UNIX_X86_ABI, function call may have stack alignment padding.
+    if (paramSize % 16 != 0)
```

Comment:
Such a workaround would be fine, but my main point was that I am not sure if this implementation is correct, since I think there is no guarantee that the esp is aligned before the callee applies the padding. And that is what you assume here.
However, if we have no such guarantee, I don't think we can recover the esp before the padding was applied unless we can somehow get the amount of padding applied by the callee. Just consider these cases (calling a function with one stack argument):
```asm
push ebp
mov ebp, esp
push ebx
// Now the esp value ends with 8
sub esp, 4
mov eax, arg
push eax
call fn
```

```asm
push ebp
mov ebp, esp
// Now the esp value ends with c
sub esp, 8
mov eax, arg
push eax
call fn
```

So you have the same size of arguments, but the padding is different, depending on how much we have pushed in the prolog. But since you consider only the size of the arguments, the ResumeEsp would be incorrect.